render_pdf/src/make_latex.py

```diff
@@ -0,0 +1,80 @@
+from typing import Union
+import parser
+import os
+import jinja2
+
+# TODO probably best to use pydantic for a config template for the dict below. 
+data = {
+        'language' : "sanskrit",
+        'title' : "",
+        'author' : "",
+        'body' : [],
+        'analysis': []
+    }
+
+latex_jinja_env = jinja2.Environment(
+        block_start_string = '<*',
+        block_end_string = '*>',
+        variable_start_string = '<<',
+        variable_end_string = '>>',
+        comment_start_string = '<=',
+        comment_end_string = '=>',
+        loader = jinja2.FileSystemLoader(os.path.abspath('.'))
+    )
+template = latex_jinja_env.get_template('template.tex')
+
+## TODO add an argument for receiving language, default is sanskrit + devanagari
+
+def generate_fields(link: str, lipi: Union["SKT","IAST"], author: str) -> tuple[list, str, str]:
+    # TODO handle lipi input
+    _ = lipi
+    #####
+
+    verses, analysis, title, parsed_author = parser.parse(url=link)
+    if parsed_author != "":
+        author = parsed_author
+    parsed_verses = []
+    parsed_analysis = []
+    
+    for verse in verses:
+        for line in verses[verse]:
+            l = [verse]
+            for segment in verses[verse][line]:
+                text = verses[verse][line][segment]
+                l.append(text)
+            parsed_verses.append(l)
+    for verse in analysis:
+        for line in verses[verse]:
+            l = [verse]
+            for segment in verses[verse][line]:
+                text = verses[verse][line][segment]
+                l.append(text)
+            parsed_analysis.append(l)
+
+    return parsed_verses, parsed_analysis, title, author
+
+def generate_texfile(link, lipi, author):
+    # TODO Author can either be passed in or extracted. Some texts don't have this. 
+    # Need to standardize input texts. 
+    body, analysis, title, author = generate_fields(link, lipi, author)
+    global data
+    data['body'] = body
+    data['title'] = title.lower()
+    data['author'] = author.lower()
+    data['analysis'] = analysis
+    if analysis == []:
+        data['hasAnalysis'] = False
+    else:
+        data['hasAnalysis'] = True
+    rendered = template.render(data)
+    with open(f"./outputs/{data['title']}-{data['author']}.tex", 'w') as f:
+        f.write(rendered)
+
+def main():
+    link = "https://raw.githubusercontent.com/ambuda-org/gretil/main/1_sanskr/tei/sa_kAlidAsa-kumArasaMbhava.xml"
```

Comment:
let's avoid any network logic here. I think the script is simplest as something that accepts two local files (input and output) as command line arguments. The simplest way to do this is:

```
import sys
input_file, output_file = sys.argv[1], sys.argv[2]
```

My indices might be off, but that's the idea.